src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java

```diff
@@ -320,9 +333,33 @@ private OkHttpClient reconfigureClientInterceptors(OkHttpClient client, String i
     if (!builder.interceptors().isEmpty()) {
       for (Iterator<Interceptor> iter = builder.interceptors().iterator(); iter.hasNext(); ) {
         Interceptor element = iter.next();
-        String currentInterceptor = element.getClass().getSimpleName();
-        if (currentInterceptor.equals(interceptorToRemove)) {
-          LOG.log(Level.FINE, "Removing interceptor " + currentInterceptor + " from http client instance.");
+        if (interceptorToRemove.equals(element.getClass().getSimpleName())) {
+          LOG.log(Level.FINE, "Removing interceptor " + element.getClass().getName() + " from http client instance.");
+          iter.remove();
+        }
+      }
+    }
+
+    return builder.build();
+  }
+
+  /**
+   * Sets a new list of interceptors for the specified {@link OkHttpClient} instance by removing any interceptors
```

Comment:
This new flavor of reconfigureClientInterceptors() was needed because we can no longer key on the class name of the interceptor class that we want to remove.   Instead, we'll remove any interceptor that implements the IRetryInterceptor marker interface.